### PR TITLE
Add SecondaryChip

### PR DESCRIPTION
**Pull request by luizgrp**
#### WHAT

Add `SecondaryChip`.

Screen Shot 2022-07-05 at 14 46 46

#### WHY

In order to have our base building blocks aligned with specs in Figma.

#### HOW

- Add implementation;
- Add previews in the debug folder;
- Add snapshot tests;

#### Checklist :clipboard:
- [x] Add explicit visibility modifier and explicit return types for public declarations
- [x] Run spotless check
- [x] Run tests
- [x] Update metalava's signature text files

### CODE REVIEW

**Review by yschimke — APPROVED** [review]
I trust you that we need it.

But we should follow up on why the defaults from MaterialChip don't work here.  What does it look like without?

**Comment by yschimke** [comment]
Essentially, if this doesn't follow material defaults, maybe they want to get closer to it. But fine if not.

**Comment by luizgrp** [comment]
@yschimke by MaterialChip do you mean [Chip](https://github.com/androidx/androidx/blob/9818b6f43b810feca64120b790df7204f0399150/wear/compose/compose-material/src/commonMain/kotlin/androidx/wear/compose/material/Chip.kt) from `androidx.wear.compose.material`?

If so, thanks for the heads up, reviewing the code I can see that I can remove few redundant attributes from the definition here, like typography and the alpha for disabled state that I [highlighted](https://github.com/google/horologist/pull/317#discussion_r913157522) is also already implemented there, so it might be an issue with the way we are creating the icon.

However, this definition here would still be beneficial, as the material Chip provides slots for the labels and icon, and our Figma definition has clear specs on truncation, max lines, text alignment etc. so all of these would have to be duplicated across all our screens that use these components. Initially I created a function just inside the screen class to be reused, but then I thought it would make sense to make it broader to be used across other screens.

See the example below the usage with this definition:

```kotlin
    SecondaryChip(
        primaryLabel = "Primary label",
        onClick = { },
        secondaryLabel = "Secondary label",
        icon = "iconUri",
        placeholder = rememberVectorPainter(
            image = Icons.Default.Add,
            tintColor = Color.White,
        )
    )
```

Whereas the same usage with material Chip would be:

```kotlin
Chip(
    label = {
        Text(
            text = "Primary label",
            modifier = Modifier
                .fillMaxWidth()
                .padding(start = 6.dp),
            textAlign = TextAlign.Left,
            overflow = TextOverflow.Ellipsis,
            maxLines = 1,
        )
    },
    onClick = { },
    modifier = modifier.fillMaxWidth(),
    secondaryLabel = {
        Text(
            text = "Secondary label",
            modifier = Modifier.padding(start = 6.dp),
            overflow = TextOverflow.Ellipsis,
            maxLines = 1,
        )
    },
    icon = {
        Image(
            painter = rememberAsyncImagePainter(model = icon, placeholder = placeholder),
            contentDescription = null, // hidden from talkback
            contentScale = ContentScale.Fit
        )
    },
    colors = ChipDefaults.secondaryChipColors(),
)
```

And we would need to be manually checking through code review that the submitted PR added those constraints to the texts/image and / or added tests to check truncation, max lines etc.

File: media-ui/src/main/java/com/google/android/horologist/media/ui/components/base/SecondaryChip.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+                    maxLines = 1,
+                    style = MaterialTheme.typography.caption2,
+                )
+            }
+        }
+
+    val iconParam: (@Composable BoxScope.() -> Unit)? =
+        icon?.let {
+            {
+                Image(
```
[the remark this answers is not in the document]
Comment (luizgrp) [reply media-ui/src/main/java/com/google/android/horologist/media/ui/components/base/SecondaryChip.kt:96]: This is already [implemented](https://github.com/androidx/androidx/blob/9818b6f43b810feca64120b790df7204f0399150/wear/compose/compose-material/src/commonMain/kotlin/androidx/wear/compose/material/Chip.kt#L226-L227) in Chip from `androidx.wear.compose.material` so we need to investigate why the Image that we are using here in the icon slot is not picking up the alpha attribute.

File: media-ui/src/main/java/com/google/android/horologist/media/ui/components/base/SecondaryChip.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                text = primaryLabel,
+                modifier = Modifier.fillMaxWidth(),
+                textAlign = if (hasSecondaryLabel || hasIcon) TextAlign.Left else TextAlign.Center,
+                overflow = TextOverflow.Ellipsis,
+                maxLines = if (hasSecondaryLabel) 1 else 2,
+            )
+        }
+
+    val secondaryLabelParam: (@Composable RowScope.() -> Unit)? =
+        secondaryLabel?.let {
```
Comment (yschimke) [inline media-ui/src/main/java/com/google/android/horologist/media/ui/components/base/SecondaryChip.kt:64]: This nullable block syntax is so awkward. But makes sense

File: media-ui/src/main/java/com/google/android/horologist/media/ui/components/base/SecondaryChip.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (luizgrp) [reply media-ui/src/main/java/com/google/android/horologist/media/ui/components/base/SecondaryChip.kt]: tracked in https://github.com/google/horologist/issues/329

**Pull request merged**